CardParts/src/Classes/Card Parts/CardPartAttributedTextView.swift

```diff
@@ -96,49 +100,62 @@ public class CardPartAttributedTextView: UIView, CardPartView {
     /// set whether the text within a text field is editable
     public var isEditable: Bool {
         didSet {
-            textView.isEditable = isEditable
+            DispatchQueue.main.async {
+                self.textView.isEditable = self.isEditable
+            }
         }
     }
     /// use to specify data that should automatically be passed toother applications as URLs, e.g. phone numbers
     public var dataDetectorTypes: UIDataDetectorTypes {
         didSet {
-            textView.dataDetectorTypes = dataDetectorTypes
+            DispatchQueue.main.async {
+                self.textView.dataDetectorTypes = self.dataDetectorTypes
+            }
         }
     }
     /// use to specify text alignment wthin the text field
     public var textAlignment: NSTextAlignment {
         didSet {
-            textView.textAlignment = textAlignment
+            DispatchQueue.main.async {
+                self.textView.textAlignment = self.textAlignment
+            }
         }
     }
     /// set attributes of linked text
     public var linkTextAttributes: [NSAttributedString.Key : Any] {
         didSet {
-            textView.linkTextAttributes = linkTextAttributes
+            DispatchQueue.main.async {
+                self.textView.linkTextAttributes = self.linkTextAttributes
+            }
         }
     }
     
     /// allows for exclusion paths to be added for text wrapping
     public var exclusionPath: [UIBezierPath]? {
         didSet {
-            let imageFrame = updateExclusionPath()
-            let exclusionPath = UIBezierPath(rect: imageFrame)
-            self.textView.textContainer.exclusionPaths = [exclusionPath]
+            DispatchQueue.main.async {
+                let imageFrame = self.updateExclusionPath()
```

Comment:
Potential memory leak - weakify self

Reply:
I thought didSet doesn't capture anything, preventing retain cycles?

Reply:
I know didSet doesn't require to weakify self but my concern was that we are now changing threads from the originally called thread... but I could very well be wrong - let me look into this a little deeper.